pygeohash/stats.py

```diff
@@ -29,41 +30,41 @@ def mean(geohashes):
     return encode(float(sum([x[0] for x in latlons])) / count, float(sum([x[1] for x in latlons])) / count)
 
 
-def northern(geohashes):
+def northern(geohashes: Iterable[str]):
     """
     Takes in an iterable of geohashes and returns the northernmost position of the group as a geohash.
 
     :param geohashes:
     :return:
     """
 
-    latlons = [decode(x) for x in geohashes]
+    latlons = (decode(x) for x in geohashes)
```

Comment:
`sorted` itself returns a `list` from any iterable, so this generator expression saves a (probably minuscule, but measurable) amount of memory/cycles by not initializing an unnecessary unsorted `list` in the interim.

Repeated this pattern below, as well